File: C#/Binary Tree Level Order Traversal/Binary Tree Level Order Traversal.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public IList<IList<int>> LevelOrder (TreeNode root) {
        IList<IList<int>> levelsVals = new List<IList<int>> ();
        if (root == null) return levelsVals;

        Queue<TreeNode> queue = new Queue<TreeNode> ();
        queue.Enqueue (root);
        while (queue.Count > 0) {
            levelsVals.Add (Helper (ref queue));
        }

        return levelsVals;
    }

    private IList<int> Helper (ref Queue<TreeNode> queue) {
        IList<int> levelVals = new List<int> ();
        int nodeCount = queue.Count;
        for (int i = 0; i < nodeCount; i++) {
            TreeNode node = queue.Dequeue ();
            levelVals.Add (node.val);
            if (node.left != null) queue.Enqueue (node.left);
            if (node.right != null) queue.Enqueue (node.right);
        }
        return levelVals;
    }
}

File: Python/Binary Tree Level Order Traversal/Binary Tree Level Order Traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None


class Solution:
    def levelOrder(self, root):
        """
        :type root: TreeNode
        :rtype: List[List[int]]
        """
        if (not root):
            return []
        queue = []
        levels_vals = []
        queue.append(root)
        while (len(queue) > 0):
            levels_vals.append(self.helper(queue))
        return levels_vals

    def helper(self, queue):
        node_count = len(queue)
        level_vals = []
        for i in range(node_count):
            node = queue.pop(0)
            if (node.left):
                queue.append(node.left)
            if (node.right):
                queue.append(node.right)
            level_vals.append(node.val)
        return level_vals
